pkg/engine/lifecycletest/step_generator_test.go

```diff
@@ -92,3 +93,107 @@ func TestDuplicateAlias(t *testing.T) {
 	_, res = TestOp(Update).Run(project, p.GetTarget(t, snap), p.Options, false, p.BackendClient, nil)
 	assert.NotNil(t, res)
 }
+
+// TestDeletedWithOptionInheritance checks that a resource that sets its parent to another resource inherits
```

Comment:
Now that the implementation is in the resource monitor, should these tests be in `step_generator_test.go` or some other file in lifecycletest?